164798252 users should be able to reset their passwords

anyatibrian opened the pull request:
### What does this PR do?

-  Enable users to be able to reset their password in case they have forgotten

### Description of tasks to be completed.

- Add tests for a password reset  functionality
- send a link to  the user's email with their password reset token
- Add pages send an email and resetting password respectively

### How should this be manually tested?

Visit the deployed application page on heroku
enter your email that you had registered with in the app,then you will recieve an email with a password reset link, follow this link that will redirect to the password reset page from where you enter and confirm your new password.

### Any background context you want to provide?

    N/A

What are the relevant pivotal tracker stories?
[#164798252](https://www.pivotaltracker.com/story/show/164798252)

### Screen shots
send password resetlink page
<img width="1182" alt="Screenshot 2019-05-16 at 19 45 23" src="https://user-images.githubusercontent.com/18057325/57872431-573ff880-7814-11e9-9d17-9db008fda1b9.png">

<img width="1281" alt="Screenshot 2019-05-16 at 19 51 58" src="https://user-images.githubusercontent.com/18057325/57872482-73dc3080-7814-11e9-8755-be155fa26a30.png">

password reset page
<img width="1137" alt="Screenshot 2019-05-16 at 19 52 54" src="https://user-images.githubusercontent.com/18057325/57872589-ae45cd80-7814-11e9-979a-3ffd2e64f037.png">

e-ian commented:
@anyatibrian I see you have three commits, please squash them into one commit

patrickf949 reviewed (commented):
@anyatibrian great work sofar,
Please resolve conflicts

e-ian reviewed (commented):
@anyatibrian, please resolve merge conflicts

e-ian commented:
@anyatibrian, check the commit message. The title should be in lower case

MuhanguziDavid reviewed (commented):
Nice work @anyatibrian , the functionality works as expected. Please ensure to:
- resolve merge conflicts
- fully test the `passwordReset.js` and `passwordResetEmail.js` files.

Commit:
ft(password-reset): users should be able to reset their passwords.

- add test for the password reset feature
- add views files for the feature
- add validations for the form fields
- add loaders for the feature

 [Finishes 164798252]

anyatibrian commented:
> Nice work @anyatibrian , the functionality works as expected. Please ensure to:
> 
>     * resolve merge conflicts
> 
>     * fully test the `passwordReset.js` and `passwordResetEmail.js` files.

am done with the project

patrickf949 reviewed (approved):
Tested
Feedback implemented
LGTM

MuhanguziDavid reviewed (changes requested):
Nice work, however, as we talked about earlier, please add a statement to dismiss previous toasts in your action so as to avoid this

<img width="369" alt="Screen Shot 2019-05-18 at 10 22 39" src="https://user-images.githubusercontent.com/38077368/57972944-ce5ac580-79a9-11e9-8ffc-fa10640c053d.png">

MuhanguziDavid commented on src/tests/components/passwordReset.test.js:
```diff
@@ -0,0 +1,38 @@
+  it("should render properly", () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+  it("should ", () => {
```
endeavour to use meaningful names for your tests

anyatibrian replied:
done

anyatibrian replied:
done

MuhanguziDavid replied:
Please elaborate on how this is done, @anyatibrian

MuhanguziDavid commented on src/tests/components/passwordResetEmail.test.js:
```diff
@@ -0,0 +1,32 @@
+    wrapper.setProps({ errors: { email: "anyatibrian@gmail" } });
+    expect(wrapper).toMatchSnapshot();
+  });
+  it("should test email errors field", () => {
```
This test and the previous one have the same name!

anyatibrian replied:
done

MuhanguziDavid commented on src/tests/components/passwordResetEmail.test.js:
```diff
@@ -0,0 +1,32 @@
+  it("should render without fail", () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+  it("should test password is too short ", () => {
```
As mentioned a number of times before, you should expect or assert something in your test

anyatibrian replied:
done

MuhanguziDavid commented on src/tests/containers/passwordResetContainer.test.js:
```diff
@@ -0,0 +1,122 @@
+    });
+  });
+
+  it("handle onclick method", () => {
```
please give meaningful names to your tests.

anyatibrian replied:
done

anyatibrian replied:
done

MuhanguziDavid commented on src/tests/containers/passwordResetContainer.test.js:
```diff
@@ -0,0 +1,122 @@
+    wrapper.setProps({ ...nextProps });
+    expect(props.history.push).toBeCalledWith("/login");
+  });
+  it("should redirect on successfully registration", () => {
```
this has the same name as the previous test

anyatibrian replied:
done

anyatibrian commented on src/styles/sidedrawer.scss:
```diff
@@ -0,0 +1,10 @@
+      border-radius: 0;
+    }
+  }
+}
```
done

anyatibrian commented:
> Nice work @anyatibrian , the functionality works as expected. Please ensure to:
> 
>     * resolve merge conflicts
> 
>     * fully test the `passwordReset.js` and `passwordResetEmail.js` files.

am done with the project

> Nice work, however, as we talked about earlier, please add a statement to dismiss previous toasts in your action so as to avoid this
> 
> <img alt="Screen Shot 2019-05-18 at 10 22 39" width="369" src="https://user-images.githubusercontent.com/38077368/57972944-ce5ac580-79a9-11e9-8ffc-fa10640c053d.png">

done

MuhanguziDavid reviewed (approved):
Awesome stuff @anyatibrian , LGTM
